Allow YOOtheme Pro integration

Skullbock opened the pull request:
See https://www.joomlaworks.net/forum/k2-en/64209-fotis-and-k2-devs-k2-as-yoothemepro-source#179948

I've added a check to see if we're in YTP to trigger the event, but IMHO the overhead of triggering an event like this is way less than doing the check, 
because it may lead to a load of the template there. Feel free to remove it.

fevangelou commented:
The code you added is out of context. You can't run this outside of a function, it's a class...

Additionally, this file is part of the backend, not the frontend.

You need to explain to me how the integration works. Is it backend only? Is it both backend AND frontend?

The more info you provide the more I can help you.

Skullbock commented:
Sorry about the out of context, my copy pasting was wrong.

I fixed that. I'm really not sure why you say this is "in the backend".
The view i'm editing here is the base view class you're using in every view class of K2, frontend and backend.

Instead of editing every single view class in the frontend, i added the piece of code in the base class, and trigger the event only on the frontend side (see the check `isClient('site')`).

fevangelou commented:
> The view i'm editing here is the base view class you're using in every view class of K2, frontend and backend.

True, my mistake.

The code does not seem to cause any issues in K2, so I'm merging it.

Out of curiosity, why are you checking that the template does not have "yootheme" in its name?

Skullbock commented:
Im checking that the name starts with yootheme because you asked me to trigger the event only on yootheme based templates. IMHO that's not necessary but...  :)

fevangelou commented:
Then it should be `!== false` ;)

Skullbock commented:
That's if I want it to contain yootheme. I want the name to **start** with yootheme. So that works. The position has to be 0.

fevangelou commented:
Correct.

fevangelou commented:
Not my finest day today, thanks for bearing with me Daniele :)

Skullbock commented:
No worries, thanks for the prompt responses and the quick help!
